File: main-api/src/models/question.js
const mongoose = require("mongoose");


const questionSchema = new mongoose.Schema({
  video_url: String,
  topic: String,
  text : String,
  helpful_tip: String,
  difficulty: String,
  category: String,
  slug : String
});

const questionModel = mongoose.model('Question', questionSchema);

module.exports = questionModel;

File: services/main_api/app/models/question.py
from beanie import Document, Indexed
from mock_interview_shared.schemas.enums import Difficulty, Category


class Question(Document):
    topic: str
    text: str
    helpful_tip: str | None = None
    difficulty: Difficulty
    category: Category
    slug: Indexed(str, unique=True)  # type: ignore[valid-type]
    video_url: str | None = None

    class Settings:
        name = "questions"
